samples/csharp/getting-started/BinaryClassification_SpamDetection/SpamDetectionConsoleApp/Program.cs

```diff
@@ -36,41 +35,36 @@ static void Main(string[] args)
+                                // While our model is relatively good at detecting the difference, this skewness leads it to always
+                                // say the message is not spam. We deal with this by lowering the threshold of the predictor. In reality,
+                                // it is useful to look at the precision-recall curve to identify the best possible threshold.
+                .Append(mlContext.Transforms.Normalize(new NormalizingEstimator.LogMeanVarColumnOptions("LogMeanVarNormalized", DefaultColumnNames.Features, useCdf: true)))
+                .AppendCacheCheckpoint(mlContext);
+
             //Create the training pipeline
             Console.WriteLine("=============== Training the model ===============");
-            var trainingPipeLine = dataProcessPipeLine.Append(mlContext.BinaryClassification.Trainers.StochasticDualCoordinateAscent());
+            var trainingPipeLine = dataProcessPipeLine.Append(mlContext.BinaryClassification.Trainers.StochasticDualCoordinateAscent(featureColumnName: "LogMeanVarNormalized"));
 
             // Evaluate the model using cross-validation.
-            // Cross-validation splits our dataset into 'folds', trains a model on some folds and 
+            // Cross-validation splits our dataset into 'folds', trains a model on some folds and
             // evaluates it on the remaining fold. We are using 5 folds so we get back 5 sets of scores.
             // Let's compute the average AUC, which should be between 0.5 and 1 (higher is better).
             Console.WriteLine("=============== Cross-validating to get model's accuracy metrics ===============");
             var crossValidationResults = mlContext.BinaryClassification.CrossValidate(data:data, estimator:trainingPipeLine, numFolds: 5);
-            var aucs = crossValidationResults.Select(r => r.metrics.Auc);
+            var aucs = crossValidationResults.Select(r => r.Metrics.Auc);
             Console.WriteLine("The AUC is {0}", aucs.Average());
 
             // Now let's train a model on the full dataset to help us get better results
             var model = trainingPipeLine.Fit(data);
 
-            // The dataset we have is skewed, as there are many more non-spam messages than spam messages.
-            // While our model is relatively good at detecting the difference, this skewness leads it to always
-            // say the message is not spam. We deal with this by lowering the threshold of the predictor. In reality,
-            // it is useful to look at the precision-recall curve to identify the best possible threshold.
-            var inPipe = new TransformerChain<ITransformer>(model.Take(model.Count() - 1).ToArray());
-            var lastTransformer = new BinaryPredictionTransformer<IPredictorProducing<float>>(mlContext, model.LastTransformer.Model, inPipe.GetOutputSchema(data.Schema), model.LastTransformer.FeatureColumn, threshold: 0.15f, thresholdColumn: DefaultColumnNames.Probability);
```

Comment:
Were you looking to remove the `threshold: 0.15f`? If there's an ok reason to, you'll want to also remove the comment which discusses the dataset skew: `The dataset we have is skewed, as there are many m...`.

Reply:
@justinormont I was attempting to denormilize the dataset  somehow to allow for the balance in the predictions. If you are okay, I was trying to still use `StochasticDualCoordinateAscent` as the original sample did. I can update the PR with your proposed pipeline. What do you think?

Reply:
There's no specific reason to use `SDCA`. `AveragedPerceptron` does better on most text datasets; I would recommend it.

The normalization that you're using won't interact with the Label Skew. In addition, the Label Skew is rather minor in this dataset (~7:1). Generally, I start to look into the skew around 10:1, and act around 20:1.

Choosing the best pipeline for an example is slightly more involved that just what's best for the given dataset as sample code is designed to demonstrate functionality and to teach best practices. 

I would purpose using my [pipeline](https://github.com/dotnet/machinelearning-samples/pull/338#discussion_r270294570), and adding another sample which actually has need of correcting the Label Skew, or has a strong business case for weighting false-positive and false-negative differently. Perhaps the Wisconsin Breast Cancer dataset, where a false-negative is significantly worse then a false-positive. Another route, would be to move the dataset used in this example.